org.jacoco.core/src/org/jacoco/core/tools/ExecDumpClient.java

```diff
@@ -123,7 +123,10 @@ public ExecFileLoader dump(final InetAddress address, final int port)
 					.setExecutionDataVisitor(loader.getExecutionDataStore());
 
 			remoteWriter.visitDumpCommand(dump, reset);
-			remoteReader.read();
+
+			if (!remoteReader.read()) {
```

Comment:
@marchof we have similar code in `org.jacoco.examples/src/org/jacoco/examples/ExecutionDataClient.java`. Later we can update it to use `ExecDumpClient`, but IMO anyway will be better to apply same modification in it with this PR.